### DISCUSSION TITLE
props problem with react Compiler and "use client"

### DISCUSSION THREAD

**Discussion by flipUp02** [vercel/next.js#65910] [App Router] [answered]
```
please fix props like arabic lang with   reactCompiler: true,

export default function MyApp() {
  return <Test text="أضف" />;

  // output is \u0623\u0636\u0641 no أضف this problem with   reactCompiler: true and with "use client" only ,  for all langs not english
}



// test.tsx
"use client"
export function Test({ text }) {
  return <div>{text}</div>;
}

```

**Comment by lubieowoce** [comment]
(cross-posted from here: https://github.com/vercel/next.js/pull/65909#issuecomment-2118455334)

I'm not sure what the problem is, the two strings seem equivalent:

<img width="240" alt="Screenshot 2024-05-18 at 00 37 33" src="https://github.com/vercel/next.js/assets/13698226/9fb0efed-b49c-4ecf-b8ff-1ef0334b4dd8">

But either way, [this is reproducible in the react compiler playground](https://playground.react.dev/#N4Igzg9grgTgxgUxALhAgHgBwjALgAgBMEAzAQygBsCSoA7OXASwjvwFkBPAQU0wAoAlPmAAdNvhgJcsNgB4AKgjAFcGXAF5RIQMRggNjBAgmDb8AegB8AbnH5TJ-NFyYoBJmHzbRoqAAYAbACYAZk8fX0DfEL8AFgBGDzoAXxAEoA), which means it's a behavior of the React Compiler, and not something specific to Next.js. If there's a problem with that output, please file an issue [in the React repo](https://github.com/facebook/react).

**Comment by flipUp02** [comment]
if use two client components with react 

check this 
https://github.com/flipUp02/nextjs

![image](https://github.com/vercel/next.js/assets/80167896/4dca7571-f42a-48e1-a9b7-d1f632a7ed25)

@lubieowoce

**Reply by lubieowoce** [reply]
Okay, I misunderstood, I thought you were talking about the string literal in the code, not about the visible output

**Reply by flipUp02** [reply]
The word should appear exactly as it does on the page "أضف" . But this does not happen. Its code is shown instead "in Unicode".

By the way, this problem is in Japanese and Chinese as well.

You can try the word 追加 in Japanese which means to add. What will appear is: \u8FFD\u52A0 not 追加

@lubieowoce

**Reply by flipUp02** [reply]
![image](https://github.com/vercel/next.js/assets/80167896/5cabc5d3-07d1-409a-9144-59bf1a9082a2)


This is the error that appears.

**Reply by flipUp02** [reply]
But without reactCompiler:true, it works fine without error

**Reply by lubieowoce** [reply, marked as answer]
This was a bug in the react compiler, and it was fixed here a couple hours ago:
https://github.com/facebook/react/pull/29141
I tested it, and your issue will be solved by upgrading to `babel-plugin-react-compiler@latest`.

Please bear in mind that the compiler is _experimental_, being constantly worked on, and possibly changing by the hour. You should always check if your problem still exists on the latest version of the compiler before reporting it, because it may have been fixed already!

**Reply by flipUp02** [reply]
@lubieowoce  thanks for help me . now it is work
